loglist/diff_check.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loglist
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"reflect"
+  "sort"
+)
+
+// warningCollector is an interface for structs that have the capability to
+// collect warnings.  It is used to report warnings from the various check
+// functions defined in this file.
+type warningCollector interface {
+	addWarning(string)
+}
+
+type warningList struct {
+	warnings []string
+}
+
+func (wl *warningList) addWarning(w string) {
+	if w != "" {
+		wl.warnings = append(wl.warnings, w)
+	}
+}
+
+// Check: operators lists are identical.
+func checkOperators(master_ll *LogList, branch_ll *LogList, wc warningCollector) {
+	if !reflect.DeepEqual(master_ll.Operators, branch_ll.Operators) {
+		wc.addWarning("Operators lists are not identical")
+	}
+}
+
+// Check: 2 logs are identical (apart from description and STH).
```

Comment:
This feels like it might be better as a method on `Log`, something like `func (l *Log) InnerEqual(other *Log) bool`.   (It's not quite an `Equal()` method because the description and STH can differ).

However, that does potentially push more work onto the user, if the detailed diffs are needed:
```golang
if !log1.InnerEqual(log2) {
  fmt.Printf("Logs differ: %+v vs %+v", log1, log2)
}
```
or:
```golang
import "github.com/kylelemons/godebug/pretty"

if !log1.InnerEqual(log2) {
  diff := pretty.Compare(log1, log2)
  fmt.Printf("Logs differ:\n%s", diff)
}
```

Reply:
I can change the func header while keeping current level of details.